### PR TITLE
[frontend] e2e tests on dashbaords - chunk 2 (#6704)

### PR DESCRIPTION
**Pull request by lndrtrbn**
### Proposed changes

* Add tests on dashboard CRUD operations

### Related issues

* #6704 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Review by marieflorescontact — COMMENTED** [review]
error when I run dashboards CRUD tests:

image

**Comment by lndrtrbn** [comment]
> error when I run dashboards CRUD tests:
> 
> image

Run insert:dev:e2e, you don't have label e2e because you have no data with this label right now

**Review by marieflorescontact — APPROVED** [review]

File: opencti-platform/opencti-front/tests_e2e/dashboard/dashboard.spec.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  const dashboardForm = new DashboardFormPage(page);
+  const widgetsPage = new DashboardWidgetsPageModel(page);
+  const malwareDetailsPage = new MalwareDetailsPage(page);
+  const dashboardDetailsPage = new DashboardDetailsPage(page);
+
+  await page.goto('/dashboard/workspaces/dashboards');
+  await leftBarPage.open();
+
+  const dashboardsMenu = 'Dashboards';
+  const dashboardName = `Dashboard - ${uuid()}`;
```
Comment (jpkha) [inline opencti-platform/opencti-front/tests_e2e/dashboard/dashboard.spec.ts:34]: Great idea 👍

File: opencti-platform/opencti-front/tests_e2e/dashboard/dashboard.spec.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  await dashboardDetailsPage.getActionsPopover().click();
+  await dashboardDetailsPage.getActionButton('Update').click();
+  await expect(dashboardForm.getUpdateTitle()).toBeVisible();
+  await dashboardForm.nameField.fill(updateDashboardName);
+  await dashboardForm.descriptionField.fill('New description');
+  await dashboardForm.getCloseButton().click();
+  await expect(dashboardDetailsPage.getTitle(updateDashboardName)).toBeVisible();
+  await dashboardDetailsPage.getActionsPopover().click();
+  await dashboardDetailsPage.getActionButton('Update').click();
+  // expect(await dashboardForm.descriptionField.value()).toEqual('New description');
```
Comment (jpkha) [inline opencti-platform/opencti-front/tests_e2e/dashboard/dashboard.spec.ts]: Should it be kept as comments?

File: opencti-platform/opencti-graphql/package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     "start:cluster": "NODE_ENV=cluster node build/back.js",
     "serv": "node build/back.js",
     "insert:dev": "node build/script-insert-dataset.js --datasets=DATA-TEST-STIX2_v2,poisonivy",
-    "insert:dev:e2e": "ELASTICSEARCH__INDEX_PREFIX=e2e node build/script-insert-dataset.js --datasets=DATA-TEST-STIX2_v2,data-test-stix-e2e",
+    "insert:dev:e2e": "ELASTICSEARCH__INDEX_PREFIX=e2e node build/script-insert-dataset.js --datasets=DATA-TEST-STIX2_v2,data-test-stix-e2e,poisonivy",
```
Comment (jpkha) [inline opencti-platform/opencti-graphql/package.json:23]: Is the script used elsewhere? Would it be better to write the dataset directly into the script?

**Review by jpkha — COMMENTED** [review]
Great PR there. Many good examples for the e2e test. Very informative 👍.

File: opencti-platform/opencti-front/tests_e2e/dashboard/dashboard.spec.ts (hunk 2 above)
Comment (lndrtrbn) [reply opencti-platform/opencti-front/tests_e2e/dashboard/dashboard.spec.ts]: Oh no indeed, I forgot this one thanks!

File: opencti-platform/opencti-graphql/package.json (hunk 3 above)
Comment (lndrtrbn) [reply opencti-platform/opencti-graphql/package.json:23]: It is the same script used for insert data for testing purpose and for local environment (cf insert:dev) that's why now we have an argument to tell which dataset to use

File: opencti-platform/opencti-front/tests_e2e/dashboard/dashboard.spec.ts (hunk 2 above)
Comment (lndrtrbn) [reply opencti-platform/opencti-front/tests_e2e/dashboard/dashboard.spec.ts]: Fixed

**Pull request merged**